### PR TITLE
Cbrom

### PR DESCRIPTION
**Pull request by cbrom**
- Basic package skeleton
- utils package
- auth package
- flag package
- api 
    - models
    - handlers
    - graphql

### CODE REVIEW

**Comment by Azrakal** [comment]
Hey @cbrom, there's a lot here for me to review (7000+ lines!), especially in one PR. In the future, it might be helpful to break out separate functionality into different wip branches so that they're easier to review, and so they can be discussed in logical blocks. For now, can you point me towards a couple of commits that you'd like me to review?

**Comment by cbrom** [comment]
Yeah, I know. This is supposed to be a skeleton for the backend, no feature in particular. 
I was hoping that you could just reveiw the structure and the general approach.

File: pkg/database/object/object.go [hunk 1]
```diff
@@ -0,0 +1,44 @@
+package object
```
Comment (Azrakal) [inline pkg/database/object/object.go]: An overall comment on this: is there any reason why you are centralizing all of your database logic in one place like this? It seems like you're duplicating a lot of what you can get from gorm out of the box. 

In my experience, it makes more sense to pass a `gorm.db` pointer around after you create it upon startup to your models, so they can use gorm directly and you don't have to write all of these queries manually.

Let me know if this doesn't make sense, I can try and explain it better.

**Comment by Azrakal** [comment]
If you can point me to 1 or 2 other big commits to review, that would be great.

**Comment by cbrom** [comment]
> I think this can just go in the handlers package. No need to be so granular.

Inside the ```user.handler.go``` file?

**Comment by Azrakal** [comment]
> 
> 
> > I think this can just go in the handlers package. No need to be so granular.
> 
> Inside the `user.handler.go` file?

Yep, I think that's fine. Reason being that these are structs directly related to the handler.

**Comment by cbrom** [comment]
> The gorm model already contains these fields. See https://gorm.io/docs/conventions.html. You can either skip using `gorm.Model` in case you really want the automatic UUID generation, or just use `gorm.Model` directly in your models.
> 
> Sorry I missed this earlier.

I believe the reason I included CreatedAt, UpdatedAt, ArchivedAt fields is that I was not using the gorm.Model embedding. But since I am using it now, it makes sense to remove these and only keep the uuid.

**Comment by Azrakal** [comment]
The gorm model contains an ID field already, so not sure if including that ID field will work or not.

**Comment by cbrom** [comment]
> The gorm model contains an ID field already, so not sure if including that ID field will work or not.

Yeah, it does work. The uint based ID is replaced with a uuid.

**Comment by Azrakal** [comment]
Looks good to me!

**Comment by cbrom** [comment]
Great! Approve and merge.

**Review by Azrakal — APPROVED** [review]

**Pull request merged**